src/shared/src/validate.rs

```diff
@@ -72,7 +72,18 @@ macro_rules! test_validate_on_deserialize {
                 }
             }
         }
+        }
     };
 }
 #[cfg(test)]
 pub(crate) use test_validate_on_deserialize;
+
+#[cfg(test)]
+pub struct TestVector<T>
```

Comment:
Another tweak: Use a standard test vector type rather than defining a dedicated test vector type for every test.